### ISSUE TITLE
Yume2kki: Black screen after Amusement Park Clown Hell Event

### ISSUE THREAD

**Issue by Ghabry** [EasyRPG/Player#1777]
Last working EasyRPG Player version: 0.5.3

---
Name of the game: Yume2kki

Download 108c JP from here: https://rmarchiv.tk/games/1905

#### Attach files (as a .zip archive or link them)

[Save04.zip](https://github.com/EasyRPG/Player/files/3221367/Save04.zip)

#### Describe the issue in detail and how to reproduce it:

Load the save, from the two doors at the top walk through the left.

Play the cutscene, when our chara gets smashed they should wake up from the dream but Player stays at a black screen.

Common Event 5 looks relevant.
Doesn't seem to be a movement issue, interpreter not stuck on "Wait for Movement".

I was expecting some stupid jumpscare or something scary because others commented on this event as "terrifying" but is really boring if you are not into such types of games.

Another report is about a black screen before a forced wakeup at the "lorn tower event", not tested, probably same force wakeup logic.

**Comment by fmatthew5876** [comment]
This is related to transitions and parallel event execution.

On Map 274, EV02 Page 2 runs parallel, turns calls EV02 Page 1, which then calls CE05 which does a erase transition.

The last commands of EV01 Page 2 + the first command in EV01 page 1 that was called enables the conditions to change EV02's active page to page 1.

When CE05 is called, it becomes the `transition_owner` (from EV02 page 2 parallel). But EV01 changes its active page so after the transition completes, the parallel worker for EV02 is never called again and so the game gets stuck with all the interpreters looping on this:

https://github.com/EasyRPG/Player/blob/master/src/game_interpreter.cpp#L206

**Comment by Ghabry** [comment]
Event trace, based on a savegame created a few frames before the actor is smashed:

```
# Session start at 2019-05-27 19:45:20
# F	frame
# E	evt_id	page_id	line_id
# M	evt_id	line_id
F	36
E	0	0	18
E	0	0	19
E	0	0	20
E	0	0	25
E	0	0	26
E	0	0	30
E	0	0	31
# 


E	0	0	33
E	0	0	34
E	0	0	35
E	0	0	36
E	0	0	37
# 


E	0	0	39
E	0	0	48
E	0	0	0
# ☆----------------------------------------------------------

｜　マップ移動中判定が ON の場合、

｜　コモンイベント『0233:同MAP内ティッシュ補』

｜　を呼び出します。
E	0	0	4
# ☆----------------------------------------------------------
E	0	0	5
E	0	0	8
E	2	2	55
# 




E	2	2	58
E	2	2	59
E	2	2	60
E	2	2	61
E	2	0	62
E	2	0	0
E	2	1	0
E	2	1	1
E	-5	0	0
F	38
E	-5	0	1
E	-8	0	0
# ・システム系の並列処理を全てOFFにして処理を軽くします。

・必ず解除とセットで使って下さい
E	-8	0	2
# 移動の切替時、キー制限、誤動作防止

ミニゲーム

画像イベントに有用かと思われ
E	-8	0	5
# ★------------------------------------------

｜
E	-8	0	7
# ｜　・メニュー禁止

｜　・ﾊﾞｰｼﾞｮﾝﾁｪｯｸｽﾙｰ

｜　・座標取得ﾅｼ＝足音ﾅｼ

│　・エフェクト効果使用不可
E	-8	0	11
# ｜

★------------------------------------------
E	-8	0	13
E	-8	0	14
E	-8	0	15
E	-8	0	16
E	-8	0	17
E	-8	0	18
E	-8	0	22
E	-8	0	23
E	-5	0	2
E	-257	0	0
E	-257	0	1
E	-257	0	2
E	-257	0	3
# <>----------------------------------------------------------

一度虫食い状態を見ていないと判定されません

<>----------------------------------------------------------
E	-257	0	6
E	-5	0	3
# ★----------------------------------------------------------
E	-5	0	4
E	-64	0	0
# ★------------------------------------------

｜
E	-64	0	2
# ｜　現実に戻る際の初期化
E	-64	0	3
# ｜

★------------------------------------------
E	-64	0	5
E	-64	0	6
E	-64	0	7
# ★------------------------------------------

｜
E	-64	0	9
# ｜　便利スイッチ、変数を初期化
E	-64	0	10
# ｜

★------------------------------------------
E	-64	0	12
E	-64	0	13
E	-64	0	14
# ★------------------------------------------

｜
E	-64	0	16
# ｜　歩行速度を初期化
E	-64	0	17
# ｜

★------------------------------------------
E	-64	0	19
E	-64	0	20
# 
```

**Comment by fmatthew5876** [comment]
This is fixed in #1740 

I haven't dug deeper, but my suspicion is the call stack refactor that doesn't use child interpreters fixes the owner problem somehow.
